api/app/Http/Controllers/Api/AuthSocialiteController.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\User;
+use App\Provider;
+use Tymon\JWTAuth\Facades\JWTAuth;
+use App\Http\Controllers\Controller;
+use Laravel\Socialite\Facades\Socialite;
+
+class AuthSocialiteController extends Controller
+{
+    private $driver = 'facebook';
+
+    public function redirectToProvider()
+    {
+        return Socialite::driver($this->driver)->stateless()->redirect();
+    }
+
+    public function handleProviderCallback()
+    {
+        $facebookUser = Socialite::driver($this->driver)->stateless()->user();
+
+        $user = $this->firstOrCreateUser($facebookUser);
+
+        $this->firstOrCreateProvider($user, $facebookUser);
+
+        $token = JWTAuth::fromUser($user);
+        $redirectUrl = sprintf('%s/login/callback/%s/%s', config('app.url_front_end'), $this->driver, $token);
+
+        return redirect()->to($redirectUrl);
+    }
+
+    private function firstOrCreateUser($facebookUser)
+    {
+        return User::firstOrCreate(['email' => $facebookUser->email], ['name' => $facebookUser->name]);
```

Comment:
We'll find the user with the e-mailadress, otherwise we'll create a user with the provider e-mailaddress and firstname.